### PR TITLE
Enable multithreading for all processes

### PR DESCRIPTION
**Pull request by vinyldarkscratch**
Cherry-picked change from the collection of changes in PR #411

### CODE REVIEW

**Comment by ryuichis** [comment]
Hey @vinyldarkscratch thanks for working on this. I am definitely not a Python export, so I had several duplicated logic that is spread across multiple files. Do you know if you could refactor the code to remove the duplications?

**Comment by vinyldarkscratch** [comment]
Glad to help out as usual.  :)  I can definitely take a shot at it, of course!  c:

- Vinyl Darkscratch-Kazotetsu, Founder of Nightwave Studios
(Sent from my Nightinfection, my iPhone)

--
If you believe you have received this email in error, please contact me at vinyldarkscratch@gmail.com, Twitter @vdscratchy, or Skype at vinyldarkscratch

File: oclint-scripts/build [hunk 1]
```diff
@@ -24,6 +24,10 @@ arg_parser.add_argument('-llvm-root', '--llvm-root', default=path.build.clang_in
 arg_parser.add_argument('-use-system-compiler', '--use-system-compiler', action="store_true", default=environment.is_linux())
 args = arg_parser.parse_args()
 
+multiple_thread = environment.cpu_count()
```
Comment (ryuichis) [inline oclint-scripts/build]: I may miss something obvious, but what's the real difference between the two?

File: oclint-scripts/ci [hunk 2]
```diff
@@ -17,8 +17,13 @@ arg_parser.add_argument('-setup', '--setup', action="store_true")
 arg_parser.add_argument('-llvm-branch', '--llvm-branch', nargs='?', choices=version.llvm_branches(), default=version.llvm_default_branch())
 arg_parser.add_argument('-release', '--release', action="store_true")
 arg_parser.add_argument('-archive', '--archive', action="store_true")
+arg_parser.add_argument('-j', type=int, default=0)
```
Comment (ryuichis) [inline oclint-scripts/ci]: Why can't we leave it default? This script is for Continuous Integration (it was once Jenkins, and now Travis CI), and maybe we can leave it to the CI environment?

File: oclint-scripts/make [hunk 3]
```diff
@@ -1,4 +1,4 @@
 #! /bin/sh -e
 
-./ci -reset
-./ci -setup -release
+./ci -reset "$@"
```
Comment (ryuichis) [inline oclint-scripts/make]: `-reset` does bunch of deletes, not sure if we really need it to run concurrently.

File: oclint-scripts/make [hunk 4]
```diff
@@ -1,4 +1,4 @@
...
+./ci -setup -release "$@"
```
Comment (ryuichis) [inline oclint-scripts/make]: This script is mainly for people who want to build oclint local releases. I still want to understand the reason to override this setting.

**Comment by vinyldarkscratch** [comment]
Basically, my reasoning for adding an override is because some people, like myself, may want to specify the number of threads the compiler should use, whether to save processing power for other apps or to tackle it with even more power.  On top of it, `environment.cpu_count()` doesn't seem to work properly on macOS, as I'm only getting "1" as the result, when I know for certain I have four cores.  (I also found out that hyperthreaded processors should use the number of logical cores instead of physical cores to achieve full power, to add a little useful information.  😉)

**Comment by ryuichis** [comment]
This pull request has been rebased into f6f59cb21624773e64af38d3bac86453da31bb9c with corresponding credit to @vinyldarkscratch .

**Pull request closed without merging**